### PR TITLE
17.05.0 rc1 changelog

### PR DESCRIPTION
**Pull request by thaJeztah**
ping @vieux @tiborvass

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,84 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Networking
+
++ Allow user to replace, and customize the ingress network [#31714](https://github.com/docker/docker/pull/31714)
+
+### Runtime
+
+- Ensure health probe is stopped when a container exits [#32274](https://github.com/docker/docker/pull/32274)
+
+### Swarm Mode
```
Comment (cpuguy83) [inline CHANGELOG.md]: https://github.com/docker/docker/pull/31144 ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,84 @@ information on the list of deprecated flags and APIs please have a look at
 https://docs.docker.com/engine/deprecated/ where target removal dates can also
 be found.
 
+## 17.05.0-ce (2017-05-03)
+
+### Builder
+
++ Add multi-stage build support [#31257](https://github.com/docker/docker/pull/31257)
```
Comment (tonistiigi) [inline CHANGELOG.md]: Link to #32063 (or both)

File: CHANGELOG.md (hunk 1 above)
Comment (vieux) [inline CHANGELOG.md]: added

File: CHANGELOG.md (hunk 2 above)
Comment (vieux) [inline CHANGELOG.md]: added

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,86 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Proceed with startup if cluster component cannot be created to allow recovering from a broken swarm setup [#31631](https://github.com/docker/docker/pull/31631)
+
+### Security
+
+* Allow setting SELinux type or MCS labels when using `--ipc=container:` or `--ipc=host` [#30652](https://github.com/docker/docker/pull/30652)
+
+
+### Deprecation
+
+- Add `--api-enable-cors` to deprecated features. This flag was marked deprecated in Docker 1.6 but not listed in deprecated features [#32352](https://github.com/docker/docker/pull/32352)
```
Comment (tiborvass) [inline CHANGELOG.md]: @vieux would prefer saying `Deprecate X` instead of `Add X to deprecated features.`

**Review by tiborvass — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (vieux) [inline CHANGELOG.md]: ok

File: CHANGELOG.md (hunk 1 above)
Comment (thaJeztah) [inline CHANGELOG.md]: ah, we forgot to add impact kageks in that one

Comment (albers) [inline CHANGELOG.md]: kageks :grin:

Comment (thaJeztah) [inline CHANGELOG.md]: LOL, typing on a phone isn't my strong point

**Comment by thaJeztah** [comment]
We need to add https://github.com/docker/docker/pull/32516 as well, if merged

File: CHANGELOG.md (hunk 3 above)
Comment (thaJeztah) [inline CHANGELOG.md]: @tiborvass updated

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,88 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Swarm Mode
+
++ Add update/rollback order for services (`--update-order` / `--rollback-order`) [#30261](https://github.com/docker/docker/pull/30261)
++ Add support for synchronous `service create` and `service update` [#31144](https://github.com/docker/docker/pull/31144)
++ Add support for "grace periods" on healthchecks through the `HEALTHCHECK --start-period` and `--health-start-period` flag to
+  `docker service create`, `docker service update`, `docker create`, and `docker run` to support containers with an initial startup
+  time [#28938](https://github.com/docker/docker/pull/28938)
+* `docker service create` now omits fields that are not specified by the user, when possible. This will allow defaults to be applied inside the manager [#32284](https://github.com/docker/docker/pull/32284)
+* `docker service inspect` now shows default values for fields that are not specified by the user [#32284](https://github.com/docker/docker/pull/32284)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Added these two (https://github.com/docker/docker/pull/32284)

File: docs/api/version-history.md [hunk 5]
```diff
@@ -23,6 +23,7 @@ keywords: "API, Docker, rcli, REST, documentation"
 * `GET /networks/(name)` now returns an `Ingress` field showing whether the network is the ingress one.
 * `GET /networks/` now supports a `scope` filter to filter networks based on the network mode (`swarm`, `global`, or `local`).
 * `POST /containers/create`, `POST /service/create` and `POST /services/(id or name)/update` now takes the field `StartPeriod` as a part of the `HealthConfig` allowing for specification of a period during which the container should not be considered unealthy even if health checks do not pass.
+* `GET /services/(id)` now accepts an `insertDefaults` query-parameter to merge default values into the service inspect output.
```
Comment (thaJeztah) [inline docs/api/version-history.md]: Relates to https://github.com/docker/docker/pull/32284

**Comment by thaJeztah** [comment]
Also need to add https://github.com/docker/docker/pull/32505 (if merged)

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,90 @@ information on the list of deprecated flags and APIs please have a look at
...
+
++ Add support for logging driver plugins [#28403](https://github.com/docker/docker/pull/28403)
+* Add support for showing logs of individual tasks to `docker service logs`, and add `/task/{id}/logs` REST endpoint [#32015](https://github.com/docker/docker/pull/32015)
+* Add `--log-opt env-regex` option to match environment variables using a regular expression [#27565](https://github.com/docker/docker/pull/27565)
+
+### Networking
+
++ Allow user to replace, and customize the ingress network [#31714](https://github.com/docker/docker/pull/31714)
+- Fix UDP traffic in containers not working after the container is restarted [#32505](https://github.com/docker/docker/pull/32505)
+- Fix files being written to `/var/lib/docker` if a different data-root is set [#32505](https://github.com/docker/docker/pull/32505)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Added these two for https://github.com/docker/docker/pull/32505 ptal @mavenugo

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,91 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Security
+
+* Allow setting SELinux type or MCS labels when using `--ipc=container:` or `--ipc=host` [#30652](https://github.com/docker/docker/pull/30652)
+
+
+### Deprecation
+
+- Deprecate `--api-enable-cors` daemon flag. This flag was marked deprecated in Docker 1.6.0 but not listed in deprecated features [#32352](https://github.com/docker/docker/pull/32352)
+- Remove Ubuntu 12.04 (Precise Pangolin) as supported platform. Ubuntu 12.04 is EOL, and no longer receives updates [#32520](https://github.com/docker/docker/pull/32520)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Added https://github.com/docker/docker/pull/32520

**Comment by thaJeztah** [comment]
Cherry-picked through https://github.com/moby/moby/pull/33606

**Pull request merged**